week3/ex2_unique_generator.py

```diff
@@ -0,0 +1,17 @@
+
+def unique(mylist):
+    dict_unique = {}
+    unique_values = [dict_unique.setdefault(e, e) for e in mylist if e not in dict_unique]
```

Comment:
This defeats whole purpose of generators. Generators are good to process data in chunks without having to keep the whole dataset in memory. It would be great, if you could rewrite this implementation so that it doesn't create intermediate lists. Just iterate over the input data and yield result as you go, no need to build a list first.

Reply:
done